### PR TITLE
fix: replace `eth_abi` with `eth_stdlib`

### LINKED ISSUE #21: bug: arrays are returned as tuples
Calling `foo()` returns `(1, 2, 3)` instead of `[1, 2, 3]`.
```vyper
@external
def foo() -> uint256[3]:
    return [1, 2, 3]
```

```
>>> boa.loads("@external\ndef foo() -> uint256[3]:\n\treturn [1,2,3]\n")
<VyperContract at 0x0000000000000000000000000000000000000067, compiled with vyper-0.3.8+847849c5>
>>> a = boa.loads("@external\ndef foo() -> uint256[3]:\n\treturn [1,2,3]\n")
>>> a.foo()
(1, 2, 3)
```

### PR DESCRIPTION
**Pull request by tserg**
Partial solution for #21 by replacing `eth_abi` with `eth_stdlib` for encoding and decoding.

One issue is that `eth_stdlib` requires python 3.10, but Vyper targets python 3.8 and above.

### CODE REVIEW

File: boa/contract.py [hunk 1]
```diff
@@ -806,10 +806,13 @@ def _prepare_calldata(self, *args, **kwargs):
         total_non_base_args = len(kwargs) + len(args) - len(self.fn_signature.base_args)
         method_id, args_abi_type = self.args_abi_type(total_non_base_args)
 
-        # allow things with `.address` to be encode-able
-        args = [getattr(arg, "address", arg) for arg in args]
+        if len(args) == 0:
```
Comment (charles-cooper) [inline boa/contract.py:809]: we don't need this branch as of https://github.com/skellet0r/eth-stdlib/pull/11

**Review by charles-cooper — COMMENTED** [review]
i was looking into this / thinking about it, and a possible solution is to use eth-stdlib optionally. so we try to import eth.codecs.abi and if it doesn't import, fall back to eth-abi.

File: boa/contract.py [hunk 2]
```diff
@@ -40,6 +39,14 @@
 from boa.vyper.decoder_utils import ByteAddressableStorage, decode_vyper_object
 
 
+try:
+    # `eth-stdlib` requires python 3.10 and above
+    from eth.codecs.abi import decode, encode
```
Comment (charles-cooper) [inline boa/contract.py:44]: can we import these with more qualified names? ex. `import decode as abi_decode, encode as abi_encode`

File: boa/contract.py [hunk 3]
```diff
@@ -201,7 +201,7 @@ def pretty_vm_reason(self):
         # decode error msg if it's "Error(string)"
         # b"\x08\xc3y\xa0" == method_id("Error(string)")
         if isinstance(err.args[0], bytes) and err.args[0][:4] == b"\x08\xc3y\xa0":
-            return decode("(string)", err.args[0][4:])[0]
+            return ("(string)", err.args[0][4:])[0]
```
[the remark this answers is not in the document]
Comment (tserg) [reply boa/contract.py]: oops sorry!

**Pull request merged**